server/mediaserver.go

```diff
@@ -711,11 +719,18 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 				}
 				s.connectionLock.RUnlock()
 				if time.Since(lastUsed) > httpPushTimeout {
-					_ = removeRTMPStream(s, mid)
+					_ = removeRTMPStream(s, intmid, extmid)
 					return
 				}
 			}
-		}(s, mid)
+		}(s, cxn.mid, mid)
+		if cxn.mid != mid {
```

Comment:
What do you think about moving this `s.internalManifests` update above the watchdog goroutine? If we do that, then `mid` would be set to the value used to index into `s.rtmpConnections` (i.e. either the push URL manifestID or the webhook manifestID if it is different) before the watchdog goroutine. Then, I think we can just pass `mid` into the watchdog goroutine and also avoid the extra argument in `removeRTMPStream()` by doing something like:

```golang
func removeRTMPStream(s *LivepeerServer, mid core.ManifestID) error {
    ...
    // Clean up s.rtmpConnections
    // Delete the internal manifestID if it exists and no-op if it doesn't
    delete(s.internalManifests, mid)
}
```

Reply:
`internalManifests` and `rtmpConnections` indexed by different values, so we need both inside `removeRTMPStream`

I should have called it `external2internalManifest` instead of `internalManifests`...

Reply:
Ah right my mistake.

- If the webhook returns a different manifestID than the push URL manifestID, rtmpConnections is indexed by the webhook manifestID and internalManifests maps push URL manifestID -> webhook manifestID
- Otherwise, rtmpConnections is indexed by the push URL manifest and internalManifests is not updated
- The push URL manifestID is the "external manifestID" and the webhook manifestID is the "internal manifestID"

Could it make sense to always pass the external manifestID to `removeRTMPStream()` and then do a lookup inside the function for the internal manifestID if it exists? Something like:

```golang
func removeRTMPStream(s *LivepeerServer, mid core.ManifestID) error {
    s.connectionLock.Lock()
    defer s.connectionLock.Unlock()
    if intmid, exists := s.internalManifests[mid]; exists {
        // Use the internal manifestID that was stored for the provided manifestID
        // to index into rtmpConnections
        mid = intmid
    }
    cxn, ok := s.rtmpConnections(mid)
    ...
}
```

This would involve an extra lookup in `removeRTMPStream()`, but IMO it feels preferable to having to pass two manifestIDs into `removeRTMPStream()` every time it is used which seems a bit cumbersome.

Reply:
Fixed in b014d7ed82ac4482071248edc58a4dcf6ef9d53c